src/App.js

```diff
@@ -83,72 +92,71 @@ function App() {
-          <NavbarRoute path="/application/:part">
-            {params => <ApplicationForm part={params.part} />}
-          </NavbarRoute>
-          <PageRoute path="/">
-            <Home />
-          </PageRoute>
-          <AuthPageRoute path="/application">
-            <Application />
-          </AuthPageRoute>
-          <PageRoute path="/charcuterie">
-            <Charcuterie />
-          </PageRoute>
-          <PageRoute path="/faq">
-            <Faq />
-          </PageRoute>
-          <PageRoute path="/schedule">
-            <Schedule />
-          </PageRoute>
-          <PageRoute path="/sponsors">
-            <Sponsors />
-          </PageRoute>
-          <PageRoute path="/quicklinks">
-            <Quicklinks />
-          </PageRoute>
-          <PageRoute path="/judging">
-            <Judging />
-          </PageRoute>
-          <PageRoute path="/judging/admin">
-            <JudgingAdmin />
-          </PageRoute>
-          <Route path="/judging/view/:id">
-            {params => (
-              <Page>
-                <JudgingView id={params.id} />
-              </Page>
-            )}
-          </Route>
-          <PageRoute path="/submission">
-            <Submission />
-          </PageRoute>
-          <PageRoute path="/submission/create">
-            <SubmissionCreate />
-          </PageRoute>
-          <PageRoute path="/submission/edit">
-            <SubmissionEdit />
-          </PageRoute>
-          <Route>
+    <ThemeProvider>
+      <GlobalStyle />
+      <Switch>
+        <PageRoute path="/">
+          <Home />
+        </PageRoute>
+        <PageRoute path="/charcuterie">
+          <Charcuterie />
+        </PageRoute>
+        <PageRoute path="/faq">
+          <Faq />
+        </PageRoute>
+        <PageRoute path="/schedule">
+          <Schedule />
+        </PageRoute>
+        <PageRoute path="/sponsors">
+          <Sponsors />
+        </PageRoute>
+        <PageRoute path="/quicklinks">
+          <Quicklinks />
+        </PageRoute>
+        <Route>
+          {/* All auth related routes should go here */}
+          <AuthProvider>
+            <NoAuthRoute path="/login">
+              <Navbar />
+              <Login />
+            </NoAuthRoute>
+            <AuthPageRoute path="/application">
+              <Application />
+            </AuthPageRoute>
+            <NavbarAuthRoute path="/application/review">
+              <ApplicationReview />
+            </NavbarAuthRoute>
+            <NavbarAuthRoute path="/application/confirmation">
+              <ApplicationConfirmation />
+            </NavbarAuthRoute>
+            <NavbarAuthRoute path="/application/:part">
+              {params => <ApplicationForm part={params.part} />}
+            </NavbarAuthRoute>
+            <AuthPageRoute path="/judging">
+              <Judging />
+            </AuthPageRoute>
+            <PageRoute path="/judging/admin">
+              <JudgingAdmin />
+            </PageRoute>
```

Comment:
@jackyzha0 this probably needs auth as well, but i'm not sure what kind so i've left it without auth for now.